packages/menu/src/Menu.tsx

```diff
@@ -35,87 +54,115 @@ interface MenuProps extends Omit<PopoverProps, 'spacing'> {
 </button>
 ```
  * @param props.children Content to appear inside of Menu.
- * @param props.active Boolean to describe whether or not Menu is active.
+ * @param props.open Boolean to describe whether or not Menu is open.
+ * @param props.setOpen Callback to change the open state of the Menu.
+ * @param props.shouldClose Callback to determine whether or not Menu should close when user tries to close it.
  * @param props.className Classname applied to Menu.
  * @param props.align Alignment of Menu relative to another element: `top`, `bottom`, `left`, `right`.
  * @param props.justify Justification of Menu relative to another element: `start`, `middle`, `end`.
  * @param props.refEl Reference element that Menu should be positioned against.
  * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in DOM tree.
  * @param props.trigger Trigger element can be ReactNode or function, and, if present, internally manages active state of Menu.
  */
+
+const escapeKey = 27;
+
 function Menu({
   align = Align.Bottom,
   justify = Justify.End,
   usePortal = true,
   adjustOnMutation = false,
-  active,
+  open = false,
+  shouldClose = () => true,
+  setOpen,
   children,
   className,
   refEl,
   trigger,
   ...rest
 }: MenuProps) {
```

Comment:
We should consider this component 'controlled' if the `open` prop is set. Have it be undefined by default and then change behavior based on if the value is given.

This is similar to how the input component behaves.

Reply:
Curious to see how this change changes the code.

Reply:
Updated so that it does not default to false, but unless I'm missing something, it doesn't change much about the code at all